src/components/createElementComponent.tsx

```diff
@@ -227,18 +161,18 @@ const createElementComponent = (
       ]);
 
       if (updates) {
-        elementRef.current.update(updates);
+        element.update(updates);
       }
-    }, [options, prevOptions]);
+    }, [options, prevOptions, element]);
 
     React.useLayoutEffect(() => {
       return () => {
-        if (elementRef.current) {
-          elementRef.current.destroy();
-          elementRef.current = null;
+        if (element) {
+          element.destroy();
+          setElement(null);
```

Comment:
can't remember the behavior of this in the later versions of react. does this throw a warning/error indicating setState being called after a component is unmounted?

Reply:
Oops, removed this line, since it's a no-op.

Interestingly, I tested rendering an element, and then after a timeout, unmounting. It seems to just work — no logs/errors.